web/server/codechecker_server/api/report_server.py

```diff
@@ -743,6 +743,12 @@ def getRunData(self, run_filter):
     def getRunHistory(self, run_ids, limit, offset, run_history_filter):
         self.__require_access()
 
+        max_query_limit = constants.MAX_QUERY_SIZE
```

Comment:
I think we could create a helper function which will check the given limit because we use it in multiple places.

In #2175 I have also used this code block.